src/Game/Client.java

```diff
@@ -2360,6 +2430,22 @@ public static float getXPforLevel(int level) {
     return xp;
   }
 
+  public static float getLevelFromXP(float xp) {
+      float lvl = 1;
+      // 136.53725 is the maximum level you can reach in RSC before XP rolls over negative
+      for (; lvl <= 137 && getXPforLevel((int)lvl) <= xp; lvl++) {}
```

Comment:
although this is allowed by compilers, I think is better practice to also include the initialization here, in the declaration of the line above can either be set to what is or 0

Reply:
Think I had done this so that lvl could have its scope slightly expanded, for use after the  loop. could still refactor lvl as an int though

Reply:
could refactor as a while loop if you would like that better lol

Reply:
yep is much better now